cpp/arcticdb/version/symbol_list.cpp

```diff
@@ -216,27 +241,31 @@ static const StreamId compaction_id {CompactionId};
         return output;
     }
 
-    void SymbolList::delete_keys(
-            std::shared_ptr<Store> store, const KeyVector& lists) {
-        std::vector<VariantKey> vars;
-        vars.reserve(lists.size());
-        for(const auto& atom_key: lists)
-            vars.emplace_back(VariantKey{atom_key});
-        store->remove_keys(vars).get();
+    folly::Future<std::vector<Store::RemoveKeyResultType>>
+    SymbolList::delete_keys(const std::shared_ptr<Store>& store, KeyVector&& to_remove, const AtomKey& exclude) {
+        std::vector<VariantKey> variant_keys;
+        variant_keys.reserve(to_remove.size());
+        for(auto& atom_key: to_remove) {
+            // Corner case: if the newly written Compaction key (exclude) has the same timestamp as an existing one
+            // (e.g. when a previous compaction round failed in the deletion step), we don't want to delete the former
+            if (atom_key != exclude) {
+                variant_keys.emplace_back(std::move(atom_key));
+            }
+        }
+
+        return store->remove_keys(variant_keys);
     }
 
-    std::optional<SymbolList::KeyVector::difference_type> SymbolList::last_compaction(const KeyVector& keys) {
+    auto SymbolList::last_compaction(const KeyVector& keys) -> std::optional<KeyVectorItr> {
```

Comment:
This pattern is not used anywhere else. Why is it used here?

Reply:
Left over from copy-and-paste of internal code